### PR TITLE
updated getAddressFunction and wallet creation

### PR DESCRIPTION
**Pull request by csirak1528**

### CODE REVIEW

File: contracts/deployer/WalletInit.sol [hunk 1]
```diff
@@ -53,14 +53,14 @@ contract WalletInit is Ownable2StepNoRenounce {
 	 * @param loginData LoginData struct
 	 * @return salt A unique salt used to initialize a funwallet
 	 */
-	function getSalt(bytes calldata loginData) public pure returns (bytes32 salt) {
```
Comment (Chazzzzzzz) [inline contracts/deployer/WalletInit.sol:56]: shall we also refactor loginData thing, given that twitter thing will be separated totally from this

File: errors.md [hunk 2]
```diff
@@ -373,7 +373,7 @@ To fix this error, make sure the permitted token is equal to the token you are t
 To fix this error, make sure to permit the token paymaster to spend your tokens
 
 ### `FW331`: TokenPaymaster.\_validatePaymasterUserOp() - the permitted transfer amount must be greater than or equal to the maxTokenCost
-
+n
```
Comment (Chazzzzzzz) [inline errors.md:376]: ?

File: contracts/deployer/WalletInit.sol [hunk 3]
```diff
@@ -53,14 +53,14 @@ contract WalletInit is Ownable2StepNoRenounce {
...
+	function getSalt(bytes calldata loginData, bytes calldata initializerCallData) public pure returns (bytes32 salt) {
 		LoginData memory data = abi.decode(loginData, (LoginData));
 		if (data.loginType == 1) {
 			// handle twitter login
 			salt = keccak256(abi.encode(data.socialHandle, data.index, data.loginType));
 		} else {
 			// handle normal EOA login
-			salt = keccak256(abi.encode(data.salt, data.loginType));
+			salt = keccak256(abi.encodePacked(keccak256(abi.encode(data.salt, data.loginType)), initializerCallData));
```
Comment (AlbertSu123) [inline contracts/deployer/WalletInit.sol:63]: Why not just keccak256(abi.encode(data.salt, data.loginType, initializerCalldata))? It seems a bit more complex to do it this way and I don't quite see the benefit

**Review by AlbertSu123 — CHANGES_REQUESTED** [review]

File: contracts/deployer/WalletInit.sol (hunk 3 above)
Comment (csirak1528) [reply contracts/deployer/WalletInit.sol:63]: agreed

**Review by AlbertSu123 — APPROVED** [review]

File: contracts/deployer/WalletInit.sol [hunk 4]
```diff
@@ -1,115 +1,42 @@
...
-	function getSalt(bytes calldata loginData) public pure returns (bytes32 salt) {
-		LoginData memory data = abi.decode(loginData, (LoginData));
-		if (data.loginType == 1) {
-			// handle twitter login
-			salt = keccak256(abi.encode(data.socialHandle, data.index, data.loginType));
-		} else {
-			// handle normal EOA login
-			salt = keccak256(abi.encode(data.salt, data.loginType));
-		}
+	function getSalt(bytes calldata loginData, bytes calldata initializerCallData) public pure returns (bytes32 salt) {
```
Comment (Chazzzzzzz) [inline contracts/deployer/WalletInit.sol:23]: do we still need loginData?

File: contracts/deployer/WalletInit.sol [hunk 5]
```diff
@@ -1,115 +1,42 @@
...
+		bytes32 baseSalt = abi.decode(loginData, (bytes32));
+		salt = keccak256(abi.encode(baseSalt, EOA_LOGIN_TYPE, initializerCallData));
```
Comment (Chazzzzzzz) [inline contracts/deployer/WalletInit.sol:25]: do we still need EOA_LOGIN_TYPE?

**Review by Chazzzzzzz — APPROVED** [review]

**Pull request merged**